app/domain/references/models/sql.py

```diff
@@ -78,16 +139,48 @@ def from_domain(cls, domain_obj: DomainReference) -> Self:
 
     def to_domain(self, preload: list[str] | None = None) -> DomainReference:
         """Convert the persistence model into a Domain Reference object."""
-        return DomainReference(
-            id=self.id,
-            visibility=self.visibility,
-            identifiers=[identifier.to_domain() for identifier in self.identifiers]
-            if "identifiers" in (preload or [])
-            else None,
-            enhancements=[enhancement.to_domain() for enhancement in self.enhancements]
-            if "enhancements" in (preload or [])
-            else None,
-        )
+        try:
+            return DomainReference(
+                id=self.id,
+                visibility=self.visibility,
+                identifiers=[identifier.to_domain() for identifier in self.identifiers]
+                if "identifiers" in (preload or [])
+                else None,
+                enhancements=[
+                    enhancement.to_domain() for enhancement in self.enhancements
+                ]
+                if "enhancements" in (preload or [])
+                else None,
+                # Note we don't propagate the opposite side of the duplicate self-join
+                # to avoid infinite recursion. Having both sides of the relationship in
+                # preload will still return the tree on both sides but won't attempt to
+                # double back.
+                canonical_reference=self.canonical_reference.to_domain(
+                    preload=[p for p in (preload or []) if p != "duplicate_references"]
+                )
+                if "canonical_reference" in (preload or []) and self.canonical_reference
+                else None,
+                duplicate_references=[
+                    reference.to_domain(
+                        preload=[
+                            p for p in (preload or []) if p != "canonical_reference"
+                        ]
+                    )
+                    for reference in self.duplicate_references
+                ]
+                if "duplicate_references" in (preload or [])
+                and self.duplicate_references
+                else None,
+                duplicate_decision=self.duplicate_decision.to_domain()
+                if "duplicate_decision" in (preload or []) and self.duplicate_decision
+                else None,
+            )
+        except MissingGreenlet as exc:
+            msg = (
+                "Trying to preload a missing relationship. This may be due to "
+                "a deeper reference duplicate depth than specified in settings."
```

Comment:
This would only be possible if we had enabled reference chaining at the dedup task, right?